plugins/module_utils/user.py

```diff
@@ -859,8 +859,8 @@ def get_impl(cursor):
     global impl
     cursor.execute("SELECT VERSION()")
     if 'mariadb' in cursor.fetchone()[0].lower():
-        from ansible_collections.community.mysql.plugins.module_utils.implementations.mariadb import user as mysqluser
-        impl = mysqluser
-    else:
         from ansible_collections.community.mysql.plugins.module_utils.implementations.mysql import user as mariauser
         impl = mariauser
+    else:
+        from ansible_collections.community.mysql.plugins.module_utils.implementations.mariadb import user as mysqluser
+        impl = mysqluser
```

Comment:
@int32bit the initial implementation is correct basically.
There's only issue with what follows after `as` and it's confusing, thanks for noticing.

Could you please adjust the PR a bit.
You should to not remove any lines, just change `if 'mariadb'` import `as mysqluser` to `as mariauser` and `impl = mysqluser` to `impl = mariauser`.
Then change things in `else` block to `as mysqluser` and `impl = mysqluser`.